src/webapp/hooks/useRoutes.ts

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { generatePath, useHistory } from "react-router-dom";
+import { FormType } from "../pages/form-page/FormPage";
+
+export enum RouteNames {
+    CREATE_FORM = "CREATE_FORM",
+    EDIT_FORM = "EDIT_FORM",
+    EVENT_TRACKER = "EVENT_TRACKER",
+    IM_TEAM_BUILDER = "IM_TEAM_BUILDER",
+    INCIDENT_ACTION_PLAN = "INCIDENT_ACTION_PLAN",
+    RESOURCES = "RESOURCES",
+    DASHBOARD = "DASHBOARD",
+}
+
+export const routes: Record<RouteNames, string> = {
+    [RouteNames.CREATE_FORM]: "/create/:formType",
```

Comment:
There is a slight mismatch between this path and the type on the inverse direction, which is more restrictive (`{ formType: FormType }`). To restrict the types with react-router-dom, the syntax is: `/create/:formType(edit|create)`. To avoid having to create the union manually (and risking making an error), consider this programmatic solution:

```
const formTypes = ["disease-outbreak-event"] as const satisfies FormType[];
const formType = `:formType(${join(formTypes, "|")})` as const;

export const routes = {
    [RouteNames.CREATE_FORM]: `/create/${formType}`,
    [RouteNames.EDIT_FORM]: `/edit/${formType}/:id`,
    ...
```

(`join` from package string-ts)

Reply:
@tokland should I add package string-ts? This could not be done with `const formType = `:formType(${formTypes.join("|")})` as const;` ?

Reply:
You can try on vscode, but I don't think string.join preserve the type. If that's the case, yes, add string-ts.

Reply:
Done!

Reply:
I had an error in workflows: 
`Error: src/webapp/hooks/useRoutes.ts(17,55): error TS1360: Type 'readonly ["disease-outbreak-event"]' does not satisfy the expected type '"disease-outbreak-event"[]'.`
`  The type 'readonly ["disease-outbreak-event"]' is 'readonly' and cannot be assigned to the mutable type '"disease-outbreak-event"[]'.`

So I have change the line to `const formTypes: FormType[] = ["disease-outbreak-event"];`

Reply:
That's interesting! It was working for me because, being a new workspace, it was using vscode's own TS (5.3.2). When configured to use the node_modules TS (5.2.2), I see this error.

However, whenever doing this kind of changes, check always the result type:

![image](https://github.com/user-attachments/assets/967761a6-3e57-4353-acd9-d5a602a88c8f)

So that's not what we need. I'd say the simplest fix is just to upgrade TS to 5.3.2.

Reply:
Thanks!! Fixed [here](https://github.com/EyeSeeTea/zebra-dev/pull/5/commits/d23914d0977503abd41476d5ed653043b5a6222e).
![Screenshot from 2024-08-05 16-00-05](https://github.com/user-attachments/assets/87aa850d-9ee2-4193-aa07-daee66ccaa73)